### PR TITLE
Feature/pa mapping

### PR DESCRIPTION
**Pull request by moltude**
ready for review

### CODE REVIEW

File: heidrun/pa_dc.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    contributor record.map( &contributor_top ).flatten
+
+    creator :class => DPLA::MAP::Agent, :each => record.field('dc:creator'), :as => :creator do
+      providedLabel creator
+    end
+
+    date :class => DPLA::MAP::TimeSpan, :each => record.field('dc:coverage'), :as => :coverage do
+      providedLabel coverage
+    end
```
Comment (guegueng) [inline heidrun/pa_dc.rb:52]: we actually don't need to map coverage to any temporal elements. Just spatial. So I think lines 50-52 can be cut.

Comment (moltude) [inline heidrun/pa_dc.rb:52]: Okay, i'll drop those lines

File: heidrun/pa_dc.rb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+  dataProvider :class => DPLA::MAP::Agent do
+    providedLabel record.field('dc:contributor').last_value
+  end
+
+  # dc:identifier
+  #  Get second occurance 
+  preview :class => DPLA::MAP::WebResource do
+   uri record.field('dc:identifier')[1..1]
+  end
```
Comment (guegueng) [inline heidrun/pa_dc.rb:95]: I think this is supposed to be isShownAt

File: heidrun/pa_dc.rb [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      providedLabel creator
+    end
+
+    date :class => DPLA::MAP::TimeSpan, :each => record.field('dc:coverage'), :as => :coverage do
+      providedLabel coverage
+    end
+
+    description record.field('dc:description')
+
+    # Selecting non-DCMIType values will be handled in enrichment <<-- CONFIRM
```
Comment (no-reply) [inline heidrun/pa_dc.rb]: :+1: this would be best handled by: https://github.com/dpla/KriKri/blob/develop/lib/krikri/enrichments/move_non_dcmi_type.rb

Comment (moltude) [inline heidrun/pa_dc.rb]: @no-reply does that change the mapping or are you just identifying the enrichment segment responsible for dropping non-DCMI types?

Comment (no-reply) [inline heidrun/pa_dc.rb]: The latter.

Though it probably means you can drop the comment (or at least the note to confirm).

Comment (moltude) [inline heidrun/pa_dc.rb]: :+1:

File: heidrun/pa_dc.rb [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+# dc:relation 
+# If there is more than one dc:relation property than return all but the first 
+# instance of the property. If one or none then return an empty set
+#
+secondary_relations = lambda do |r|
+  if(r['dc:relation'].count > 1)
+    r['dc:relation'].last_value( r['dc:relation'].count-1 )
+  else 
+    # TODO there is probably a better way return an empty set that isn't nil
```
Comment (no-reply) [inline heidrun/pa_dc.rb]: I'm seeing you need this to `#flatten`.  I'm wondering if just `[]` doesn't work?

File: heidrun/pa_dc.rb [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+#   If one or zero exist then do nothing
+# 
+# Feels kludgey as heck, should rewrite once ValueArray supports array sliceing 
+# See ticket https://issues.dp.la/issues/8287
+contributor_top = lambda do |r|
+  # testing 
+  # r['dc:contributor'].first_value( r['dc:contributor'].count-1 )
+  # end test
+
+  if r['dc:contributor'].count > 1
```
Comment (no-reply) [inline heidrun/pa_dc.rb]: This approach looks okay, as you say, until we close 8287.

You might avoid running the `#['dc:contributor']` and `#count` logic repeatedly with something like:

``` ruby
contrib = r['dc:contributor']
count   = contrib.count

(count > 1) ? contrib.first_value(count - 1) : contrib 
# ^ or do this as `if; else; end`, if preferred
```

and similarly for the example below.

**Comment by moltude** [comment]
This is ready for to merge and deploy to production @guegueng @anarchivist

**Comment by anarchivist** [comment]
:squash: ?

**Comment by moltude** [comment]
@anarchivist will do

**Comment by moltude** [comment]
:squash:ed

**Pull request merged**